src/store/reducers/subjects.js

```diff
@@ -44,7 +49,11 @@ const subjects = (state = { primary: [], lingual: [] }, action) => {
               ...subject,
               name: action.payload.name,
               elementaryScore: action.payload.elementaryScore,
-              advancedScore: action.payload.advancedScore
+              advancedScore: action.payload.advancedScore,
+              max: getSubjectScore({
```

Comment:
This `max` value should be computed in `updateSubject` action creator, so it provides all data which will be saved.